x-pack/filebeat/input/awss3/sqs.go

```diff
@@ -79,7 +79,11 @@ func (r *sqsReader) Receive(ctx context.Context) error {
 		// Process each SQS message asynchronously with a goroutine.
 		r.log.Debugf("Received %v SQS messages.", len(msgs))
 		r.metrics.sqsMessagesReceivedTotal.Add(uint64(len(msgs)))
+		// Initialize the sqs_message_waiting_guage to 0 to indicate that that SQS messages are received.
+		// PollSqsWaitingMetric shall reassign the value if there are messages waiting or if there is an error in processing the messages.
+		r.metrics.sqsMessagesWaiting.Set(int64(0))
```

Comment:
Why is this change necessary? I see the comment, but I don't understand why we need "to indicate that that SQS messages are received" via this metric. That gives the metric more that one purpose.

Reply:
I think the comment is mis-phrased. The scenario I thought was
- `sqs_message_waiting_gauge` is initialized with a value of `-1`
- [PollSqsWaitingMetric](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/awss3/input.go#L411) function checks and updates this metric every minute , but if the SQS messages are received and processed in less than a minute, then this metric is never updated? Then if this metric is polled, the value will be returned as `-1` which gives wrong interpretation. So , this line would set a value to `0` if the SQS messages are successfully received and the next step would be to invoke individual `go routines` to process each SQS message.

Do you think there could be a better way to achieve this?

Reply:
I think, this can be moved to [PollSqsWaitingMetric](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/awss3/input.go#L411) function to initialize the value to `0` just at the start of `one minute` time interval.